lib/ansible/modules/network/checkpoint/checkpoint_access_rule.py

```diff
@@ -240,7 +242,8 @@ def main():
                 pass
         elif code == 404:
             code, response = create_access_rule(module, connection)
-
+            if code != 200:
```

Comment:
Don't need this if statement, but I also don't know that this is the correct place for that failure. At this point we know `code` isn't `200` because the if statement to get there is `elif code == 404`

Reply:
@maxamillion this check has been added to handle the error response scenarios from fn. `create_access_rule` which internally calls checkpoint api `add-access-rule` and only success response of this api will result to `200` rest all will be error response and to handle those error response I have added the respective check. In the case of `success` scenario, the module shall execute without any issue. _(ref: ahttps://sc1.checkpoint.com/documents/latest/APIs/index.html?#web/add-access-rule~v1.4%20)_